halo2_proofs/src/plonk/vanishing/prover.rs

```diff
@@ -46,18 +46,24 @@ impl<C: CurveAffine> Argument<C> {
         domain: &EvaluationDomain<Scheme::Scalar>,
         mut rng: R,
         transcript: &mut T,
-    ) -> Result<Committed<Scheme::Curve>, Error> {
-        // Sample a random polynomial of degree n - 1
+    ) -> Result<Committed<Scheme::Curve, ZK>, Error> {
         let mut random_poly = domain.empty_coeff();
-        for coeff in random_poly.iter_mut() {
-            *coeff = C::Scalar::random(&mut rng);
+
+        // Sample a random polynomial of degree n - 1
+        if ZK {
+            for coeff in random_poly.iter_mut() {
+                *coeff = C::Scalar::random(&mut rng);
+            }
```

Comment:
Using `iter.for_each` I think we could take advantage of `autovectorisation` in Rust and squeeze a bit more performance due to internal compiler parallelization addition.

In other places where iterators are much more tricky than `for` loops I wouldn't mention it. But here is pretty straightforward.

Reply:
I see, but this part already exists before this PR, and I choose to not change any algorithm if not necessary, otherwise I'm afraid to make this PR more messy.

Reply:
Makes sense then. 
Thanks for the clarification :)